File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    obj = {}
    
    for index, item in enumerate(s):
        if item not in obj:
            obj[item] = index
            answer.append(-1)
        else:
            answer.append(index - obj[item])
            obj[item] = index
            
    
    return answer

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    const answer = [];
    const stack = [];
    
    for (const item of s) {
        const targetIndex = stack.indexOf(item);
        if (targetIndex === -1) answer.push(-1);
        else {
            answer.push(targetIndex + 1)
        }
        
        stack.push(item);
    }
    return answer;
}